core/src/main/scala/org/apache/spark/status/storeTypes.scala

```diff
@@ -286,6 +289,7 @@ private[spark] class TaskDataWrapper(
       taskId,
       index,
       attempt,
+      partitionId,
       new Date(launchTime),
```

Comment:
+CC @thejdeep, please take a look to make sure there are no backwardly incompatible changes.
Existing event files and/or level db's should be readable after this change.

Reply:
I have test it with event files, leveldb and rocksdb, and it is ok with rocksdb and event files, which retruns `"partitionId" : -1,`. But when testing leveldb, we get `"partitionId" : 0,`, which is different from what we expected.

BTW, the test scenario is generating eventLog and leveldb/rocksdb with old spark-client and read it on new spark-client which has this pr.

Reply:
Would be good to see if we can default the value to -1 if missing.
Else cleaning up all level-db would be a prerequisite to leveraging this.

I am not sure what is the cleanest way to enforce incompatible leveldb format changes - IIRC you had made some changes recently which require regen, right @dongjoon-hyun ? Any suggestions ? (IIRC we did not bump up the store version)

Reply:
ping @dongjoon-hyun

Reply:
Yet to hear from @dongjoon-hyun on it ... do you have any thoughts @Ngone51  ?

Reply:
@Ngone51 Thanks for your attension. You can get the actual `partitionId` from [rest-api](https://spark.apache.org/docs/latest/monitoring.html#rest-api), such as `/applications/[app-id]/stages/[stage-id]/[stage-attempt-id]/taskList`.

Reply:
leveldb is used by default right? If so, I did see partitionId is -1 actually:
<img width="792" alt="WeChat05d412b13735d2fe389b4c1199307fe1" src="https://user-images.githubusercontent.com/16397174/159113195-ac7a00f8-27bb-4c95-883a-49a108c0318e.png">

Could you double check again? @jackylee-ch 
.

Reply:
> Could you double check again? @jackylee-ch .

@Ngone51 Thanks for your double check. It works fun while reading old eventLog json files with new spark-client. The recurring scenario of this problem is that using the new spark-client read old leveldb data.
```
"7" : {
      "taskId" : 7,
      "index" : 1,
      "attempt" : 0,
      "partitionId" : 0,
      ...
}
```

Reply:
Could you give the reproduce steps? @jackylee-ch

Reply:
Firstly, use old spark-client to generate eventLog and leveldb with these configs;
```
spark.history.fs.logDirectory       /tmp/spark-client/spark-old/eventLog
spark.history.store.path            /tmp/spark-client/spark-old/kvstore
spark.history.store.hybridStore.diskBackend     LEVELDB
```
Then, use new spark-client to read eventLog and level db with the configs in step 1;

These are the reproduce steps and configs that I have used. @Ngone51

Reply:
Thanks, I have reproduced the issue with the above steps. And now I see where the problem is. Basically, the fix in `JsonProtocol` only solves the compatible issue when reading event files. But, when reading from LeveDB, there's no compatible conversion for it. I tried to set the `partitionId` of `TaskDataWrapper` with the default value -1 and seems to work. Could you double-check? @jackylee-ch

Reply:
@Ngone51 Thanks for your advise. It works fun with default value. I will add the default value -1 and some comments to it.